osu.Game/Rulesets/Replays/FramedReplayInputHandler.cs

```diff
@@ -97,7 +98,7 @@ protected FramedReplayInputHandler(Replay replay)
         {
             // TODO: This replay frame ordering should be enforced on the Replay type.
             // Currently, the ordering can be broken if the frames are added after this construction.
-            replay.Frames.Sort((x, y) => x.Time.CompareTo(y.Time));
+            replay.Frames = replay.Frames.OrderBy(f => f.Time).ToList();
```

Comment:
Note that if a ruleset stores a reference to the `replay.Frames`, it changes behavior (mutation to the list no longer applied).